pkg/config/config.go

```diff
@@ -28,4 +28,11 @@ const (
 	// RolloutDownscaleLeaderAnnotationKey is the name of the leader statefulset that should be used to determine
 	// the number of replicas in a follower statefulset.
 	RolloutDownscaleLeaderAnnotationKey = "grafana.com/rollout-downscale-leader"
+
+	RolloutMirrorReplicasFromResourceAnnotationKeyPrefix = "grafana.com/rollout-mirror-replicas-from-resource"
+	// RolloutMirrorReplicasFromResourceNameAnnotationKey -- when set (together with "kind" and optionally "api-version" annotations), rollout-operator sets number of
+	// replicas based on replicas in this resource (its scale subresource).
+	RolloutMirrorReplicasFromResourceNameAnnotationKey       = RolloutMirrorReplicasFromResourceAnnotationKeyPrefix + "-name"
+	RolloutMirrorReplicasFromResourceKindAnnotationKey       = RolloutMirrorReplicasFromResourceAnnotationKeyPrefix + "-kind"
```

Comment:
[non blocking] Have you considered a simpler `grafana.com/rollout-mirror-replicas-from-resource: ResourceTemplate/ingester-zone-a` where kind and name are separated by `/`? `kubectl` supports the same, so should be safe.

Reply:
This follows KEDA ScaledObject's `spec.ScaleTargetRef` and HPA's `scaleTargetRef` fields ([doc](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#crossversionobjectreference-v2-autoscaling)), and I copied some code for handling this stuff from HPA codebase. I'd prefer to keep it as-is for now.

Reply:
Yes, sure.